lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py

```diff
@@ -285,22 +278,33 @@ def _right_has_values_of_left(left, right):
 
                             for list_val in left_list:
                                 if list_val not in right_list:
+                                    # if list_val is the port mapping, the key 'protocol' may be absent (but defaults to 'tcp')
+                                    # fill in that default if absent and see if it is in right_list then
+                                    if isinstance(list_val, dict) and not list_val.get('protocol'):
+                                        modified_list_val = dict(list_val)
+                                        modified_list_val.update(protocol='tcp')
+                                        if modified_list_val in right_list:
+                                            continue
                                     return False
                         else:
                             return False
 
                 # Make sure right doesn't have anything that left doesn't
                 for k, v in right.items():
                     if v and k not in left:
-                        return False
+                        # 'essential' defaults to True when not specified
+                        if k == 'essential' and v is True:
+                            pass
+                        else:
+                            return False
 
                 return True
 
             def _task_definition_matches(requested_volumes, requested_containers, existing_task_definition):
                 if td['status'] != "ACTIVE":
                     return None
 
-                existing_volumes = td.get('volumes', []) or []
```

Comment:
This is actually useful, since a `.get('volumes', [])` can still return `None` if the value `volumes` is explicitly None, as opposed to nonexistent.